awx/ui/src/screens/Job/JobOutput/useJobEvents.js

```diff
@@ -24,7 +26,14 @@ export default function useJobEvents(callbacks, isFlatMode) {
   };
   const reducer = jobEventsReducer(callbacks, isFlatMode, enqueueAction);
   const [state, dispatch] = useReducer(reducer, initialState);
-
+  useEffect(() => {
+    if (state.isAllCollapsed) {
+      dispatch({
+        type: TOGGLE_COLLAPSE_ALL,
+        isCollapsed: state.isAllCollapsed,
+      });
+    }
+  }, [state.isAllCollapsed, state.events]);
```

Comment:
this seems inefficient. You re-run the full toggle all algorithm each time events are added to the tree. I don't think this would be necessary if you set the new node's `isCollapsed` flag equal to `isAllCollapsed` each time you add a new node (line 185)